express_relay/programs/express_relay/src/lib.rs

```diff
@@ -0,0 +1,366 @@
+
+        Ok(())
+    }
+
+    // Sets a new set of fee splits for the program
+    pub fn set_splits(ctx: Context<SetSplits>, data: SetSplitsArgs) -> Result<()> {
+        validate_fee_split(data.split_protocol_default)?;
+        validate_fee_split(data.split_relayer)?;
+
+        let express_relay_metadata_data = &mut ctx.accounts.express_relay_metadata;
+
+        express_relay_metadata_data.split_protocol_default = data.split_protocol_default;
+        express_relay_metadata_data.split_relayer = data.split_relayer;
+
+        Ok(())
+    }
+
+    // Sets a fee split for a specific protocol
+    pub fn set_protocol_split(ctx: Context<SetProtocolSplit>, data: SetProtocolSplitArgs) -> Result<()> {
+        validate_fee_split(data.split_protocol)?;
+
+        ctx.accounts.protocol_config.protocol = *ctx.accounts.protocol.key;
+        ctx.accounts.protocol_config.split = data.split_protocol;
+
+        Ok(())
+    }
+
+    // Permissions a particular (protocol, permission) pair and distributes bids according to splits
+    pub fn permission(ctx: Context<Permission>, data: PermissionArgs) -> Result<()> {
+        if data.deadline < Clock::get()?.unix_timestamp {
+            return err!(ErrorCode::DeadlinePassed);
+        }
+
+        // check that not cpi
+        if get_stack_height() > TRANSACTION_LEVEL_STACK_HEIGHT {
+            return err!(ErrorCode::InvalidCPIPermission);
+        }
+
+        // handle bid payment
+        let bid_amount = data.bid_amount;
+        let searcher = &ctx.accounts.searcher;
+        let rent_searcher = Rent::get()?.minimum_balance(searcher.to_account_info().data_len());
+        if bid_amount + rent_searcher > searcher.lamports() {
+            return err!(ErrorCode::InsufficientSearcherFunds);
+        }
+
+        let express_relay_metadata = &ctx.accounts.express_relay_metadata;
+        let split_relayer = express_relay_metadata.split_relayer;
+        let split_protocol_default = express_relay_metadata.split_protocol_default;
+
+        let split_protocol: u64;
+        let protocol_config = &ctx.accounts.protocol_config;
+        let protocol_config_account_info = protocol_config.to_account_info();
+        if protocol_config_account_info.data_len() > 0 {
+            let account_data = &mut &**protocol_config_account_info.try_borrow_data()?;
+            let protocol_config_data = ConfigProtocol::try_deserialize(account_data)?;
+            split_protocol = protocol_config_data.split;
+        } else {
+            split_protocol = split_protocol_default;
+        }
+
+        let fee_protocol = bid_amount * split_protocol / FEE_SPLIT_PRECISION;
+        if fee_protocol > bid_amount {
+            return err!(ErrorCode::FeesHigherThanBid);
+        }
+
+        let fee_relayer = bid_amount.saturating_sub(fee_protocol) * split_relayer / FEE_SPLIT_PRECISION;
+        if fee_relayer.checked_add(fee_protocol).ok_or(ProgramError::ArithmeticOverflow)? > bid_amount {
```

Comment:
I think this condition can only trigger if the splits are > 1, which I think is guarded against when the configuration is set. Both fee_relayer and fee_protocol are truncated, i.e., rounded down, so there's no way for them to be larger than the bid.

It's fine to leave this check here to be extra safe, but it's worth leaving a comment that this is defensive programming and really shouldn't happen.